File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
    vector<int>dp;
public:
    int climbStairs(int n) {
        dp.assign(n+1,-1);
     return dfs(n);

        
    }

    int dfs(int n)
    {
        if(n==0) return 1;
        if(n<0) return 0;

        if(dp[n]!=-1) return dp[n];

        return dp[n]=dfs(n-1)+dfs(n-2);
    }
};

File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        #dfs i will have to do, it could b

        dp=defaultdict(int)


        def dfs(i)->int:
            if i==0:
                return 1
            if i<0:
                return 0
            if i in dp:
                return dp[i]
            dp[i]=dfs(i-1) + dfs(i-2)
            return dp[i]
        return dfs(n)
        
            

        